x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/exporter/Exporter.java

```diff
@@ -102,9 +102,23 @@ public void validate(final String value, final Map<Setting<?>, Object> settings)
     /**
      * Every {@code Exporter} allows users to use a different index time format.
      */
-    private static final Setting.AffixSetting<String> INDEX_NAME_TIME_FORMAT_SETTING =
+    static final Setting.AffixSetting<String> INDEX_NAME_TIME_FORMAT_SETTING =
             Setting.affixKeySetting("xpack.monitoring.exporters.","index.name.time_format",
-                    key -> Setting.simpleString(key, Property.Dynamic, Property.NodeScope));
+                    key -> Setting.simpleString(
+                        key,
+                        new Setting.Validator<String>() {
+                            @Override
+                            public void validate(String value) {
+                                try {
+                                    DateFormatter.forPattern(value).withZone(ZoneOffset.UTC);
+                                } catch (IllegalArgumentException e) {
```

Comment:
I would suggest to open this up to catch RuntimeException so that validation (not cluster state application) catches the error.